persistence/src/main/java/org/cbioportal/genome_nexus/persistence/internal/EnsemblRepositoryImpl.java

```diff
@@ -109,4 +111,34 @@ public EnsemblTranscript findOneByHugoSymbolIgnoreCase(String hugoSymbol, String
         }
         return null;
     }
+
+    @Override
+    public HashMap<String, String> getHugoSymbolToEntrezGeneIdMap() {
+        HashMap<String, String> map = new HashMap<>();
+        List<EnsemblCanonical> transcripts = mongoTemplate.findAll(EnsemblCanonical.class, CANONICAL_TRANSCRIPTS_COLLECTION);
+        for (EnsemblCanonical transcript : transcripts) {
+            String[] previousSymbols = transcript.getPreviousSymbols();
+            String[] synonyms = transcript.getSynonyms();
+
+            if (!map.containsKey(transcript.getHugoSymbol())) {
+                map.put(transcript.getHugoSymbol(), transcript.getEntrezGeneId());
+            }
+            if (previousSymbols != null) {
+                for (String previousSymbol : previousSymbols) {
+                    if (!map.containsKey(previousSymbol)) {
+                        map.put(previousSymbol, transcript.getEntrezGeneId());
+                    }
+                }
+            }
+            if (synonyms != null) {
+                for (String synonym : synonyms) {
+                    if (!map.containsKey(synonym)) {
```

Comment:
same comment as above. We don't take any action if there is already an entry for the same symbol, so we can simply override it for now.